### PR TITLE
feat(optimizer): add cardinality trait

### PR DESCRIPTION
**Pull request by MingjiHan99**
Signed-off-by: MingjiHan <mingji_han@outlook.com>

We may need an execution context (like DuckDB) to provide statitsics in storage layer for optimizer @skyzh .

### CODE REVIEW

File: src/optimizer/plan_nodes/logical_limit.rs [hunk 1]
```diff
@@ -32,6 +32,10 @@ impl LogicalLimit {
     pub fn limit(&self) -> usize {
         self.limit
     }
+    #[allow(dead_code)]
+    fn estimated_cardinality(&self) -> usize {
```
Comment (skyzh) [inline src/optimizer/plan_nodes/logical_limit.rs:36]: should be implemented on `PlanNode`?

File: src/optimizer/plan_nodes/physical_projection.rs [hunk 2]
```diff
@@ -21,6 +21,10 @@ impl PhysicalProjection {
     pub fn logical(&self) -> &LogicalProjection {
         &self.logical
     }
+    #[allow(dead_code)]
+    fn estimated_cardinality(&self) -> usize {
```
Comment (skyzh) [inline src/optimizer/plan_nodes/physical_projection.rs:25]: ditto?

**Review by skyzh — APPROVED** [review]
Rest LGTM, good work! It's true that we might expose the storage context to the optimizer, so that we can call storage's statistics interface.

File: src/optimizer/plan_nodes/logical_limit.rs (hunk 1 above)
Comment (xxchan) [reply src/optimizer/plan_nodes/logical_limit.rs:36]: Why this comment is not resolved...?

**Pull request merged**